File: src/main/java/hu/elte/alkfejl/receptek/model/Recept.java
package hu.elte.alkfejl.receptek.model;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;



import javax.persistence.*;
import java.util.List;

@Entity
@Data
public class Recept {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Integer id;

    @Version
    private int version;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String text;


    @ManyToOne
    @JoinColumn
    private User user;

    @ElementCollection
    private List<String> alapanyagok;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    public enum Status {
        PUBLIC, PRIVATE
    }


}

File: spa/RECEPTEK/src/app/model/Recept.ts
import {User} from "./User";

export class ReceptStatus {
  static PUBLIC: String = "PUBLIC";
  static PRIVATE: String = "PRIVATE";
  static PENDING: String = "PENDING";
}

export class Recept {
  id: number;
  user: User;
  name: String;
  text: String;
  status: String;
  bevasarlolista: String[];

  constructor(name?: String, text?: String, status?: String, user?: User, id?: number) {
    this.user = user;
    this.name = name;
    this.status = status;
    this.text = text;
    this.id = id;
  }
}
